packages/back-end/src/integrations/Snowflake.ts

```diff
@@ -43,60 +34,17 @@ export default class Snowflake extends SqlIntegration {
   ensureFloat(col: string): string {
     return `CAST(${col} AS DOUBLE)`;
   }
-  async getInformationSchema(): Promise<InformationSchema[]> {
-    const database = this.params.database;
-
-    if (!database) {
+  getInformationSchemaFromClause(): string {
+    if (!this.params.database)
       throw new Error(
         "No database provided. In order to get the information schema, you must provide a database."
       );
-    }
-
-    const queryString = `SELECT
-        table_name,
-        table_catalog,
-        table_schema,
-        count(column_name) as column_count
-    FROM
-        ${database}.INFORMATION_SCHEMA.COLUMNS
-    WHERE
-        table_schema
-      NOT IN ('INFORMATION_SCHEMA')
-    GROUP BY (table_name, table_schema, table_catalog)`;
-
-    const results = await this.runQuery(queryString);
-
-    if (!results.length) {
-      throw new Error(`No tables found for database "${database}".`);
-    }
-
-    return formatInformationSchema(
-      results as RawInformationSchema[],
-      "snowflake"
-    );
+    return `${this.params.database}.information_schema.columns`;
   }
-
-  async getTableData(
-    databaseName: string,
-    tableSchema: string,
-    tableName: string
-  ): Promise<{ tableData: null | unknown[]; refreshMS: number }> {
-    const sql = `SELECT
-          data_type,
-          column_name
-        FROM
-          ${databaseName}.INFORMATION_SCHEMA.COLUMNS
-        WHERE
-          table_name
-        IN ('${tableName}')
-        AND
-          table_schema
-        IN ('${tableSchema}')`;
-
-    const queryStartTime = Date.now();
-    const tableData = await this.runQuery(sql);
-    const queryEndTime = Date.now();
-
-    return { tableData, refreshMS: queryEndTime - queryStartTime };
+  getInformationSchemaWhereClause(): string {
+    return "table_schema NOT IN ('INFORMATION_SCHEMA')";
```

Comment:
This is a bit of an odd one. According to Snowflake they use `Standard (case-sensitive) SQL comparison semantics. Snowflake automatically converts unquoted, case-insensitive identifiers to uppercase internally, so an unquoted object name appears in upper case in the Information Schema views.`

I say it's weird because in the `getInformationSchemaFromClause` clause above, everything is lowercase, and in the `getInformationSchemaTableFromClause` below, we use lowercase as well, and it works as expected.

However, with this clause, if `INFORMATION_SCHEMA` is lowercased, we are showing two schemas - one for `PUBLIC` and another for `INFORMATION_SCHEMA`.